node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -1379,65 +1401,6 @@ async fn dequeue_next_collation_and_fetch<Context>(
 	}
 }
 
-/// Handle a fetched collation result.
-#[overseer::contextbounds(CollatorProtocol, prefix = self::overseer)]
-async fn handle_collation_fetched_result<Context>(
-	ctx: &mut Context,
-	state: &mut State,
-	(mut collation_event, res): PendingCollationFetch,
-) {
```

Comment:
First, it's better to keep this function for a better code granularity.
Secondly, notice that it tries to fetch another collation in case of a failure, your code doesn't.
This logic is quite important and easy to miss, so please evaluate such scenarios.

Reply:
> First, it's better to keep this function for a better code granularity.

I agree that the function has become quite lengthy now, but the reason why it was split beforehand is that we had additional `oneshot::channel`s for collation fetch results that were used as an intermediate buffer (but I removed it since it was a useless indirection) . Both functions would now be semantically part of the same operation: handling collation fetch responses. So I would find it quite counter-intuitive to have a split.

> Secondly, notice that it tries to fetch another collation in case of a failure, your code doesn't. This logic is quite important and easy to miss, so please evaluate such scenarios.

This specific case is a class of errors that was completely removed after this refactor. The error you mentioned is a `oneshot::Canceled`. Since the `channel` is no longer present, this error doesn't exist any more.

Reply:
You're right, the bug I mentioned is actually present in master and was fixed in async backing branch.